pypit/arcomb.py

```diff
@@ -32,16 +32,22 @@ def comb_frames(frames_arr, det, frametype, **kwargs):
     method = settings.argflag[frametype]['combine']['method']
     satpix = settings.argflag[frametype]['combine']['satpix']
     saturation = settings.spect[dnum]['saturation']*settings.spect[dnum]['nonlinear']
-    return core_comb_frames(frames_arr, frametype=frametype,
-                method=method, reject=reject, satpix=satpix, saturation=saturation, **kwargs)
+    return core_comb_frames(frames_arr, frametype=frametype, saturation=saturation,
+                            method=method, satpix=satpix, cosmics=reject['cosmics'],
+                            n_lohi=reject['lowhigh'], sig_lohi=reject['level'],
+                            replace=reject['replace'], **kwargs)
 
-def core_comb_frames(frames_arr, maskvalue=1048577, printtype=None, frametype='Unknown',
-                method='weightmean', reject=None, satpix='reject', saturation=None):
-    """ Combine several frames
+
+def core_comb_frames(frames_arr, printtype=None, frametype='Unknown', saturation=None,
+                     maskvalue=1048577, method='weightmean', satpix='reject', cosmics=None,
+                     n_lohi=[0,0], sig_lohi=[3.,3.], replace='maxnonsat'):
+    """
+    Combine several frames
 
     .. todo::
         - Make better use of np.ma.MaskedArray objects throughout?
         - More testing of replacement code necessary?
+        - Improve docstring...
```

Comment:
Do you have specific examples?  I've used masked arrays quite a bit and I have had issues now and again.  But those issues have mostly (always?) been because I've assumed some function understands what a masked array is when it doesn't.  When sticking to the MaskedArray class itself and its methods, I've been safe.  Personally, I think the gains from using MaskedArrays is worth the care one has to take when using them (i.e., checking to make sure that they're behaving as you expect/assume).

Reply:
Related, please see Issue #361

Reply:
Not sure if this is the right time to deal with this, though.
Should probably be its own PR.

Reply:
Hi @kbwestfall, 

I had many problems with numpy masked arrays in the past. Whether that was due to buggy code, or user error, i.e. the fact that I thought a numpy routine was respecting the mask, whereas it actually was not, is not clear. 

The point is that masking is an important and subtle issue in spectroscopic data reduction. We have several masks and we mask things on multiple occasions and constantly update that mask.  There is no way that this masking can be implicit like with numpy arrays, because it
will result in code that is unreadable and it will cause errors. 

In my opinion we should remove all usage of them from PYPIT and just propagate around an extra mask when needed.

Reply:
Agreed that pixels can be masked for a number of reasons.  I have a convience `BitMask` class that I found helpful when I was doing this for MaNGA.

Personally, I find MaskedArrays very valuable and have made extensive use of them, but I'm fine with discussing this more.  Let's please move this discussion to #361 or a different, new issue.